scripts/consume_sitemaps.py

```diff
@@ -127,33 +125,36 @@ def pre_thing_host_url(thing_url: str) -> typing.Optional[str]:
 def construct_thing_futures(
     thing_futures: list,
     sitemap_file_iterator: Iterator,
+    sitemap_file_url: str,
     rsession: requests.sessions,
     thing_executor: ThreadPoolExecutor,
 ) -> bool:
     constructed_all_futures_for_sitemap_file = False
-    thing_ids = set()
-    things_url = None
-    while len(thing_ids) < CONCURRENT_DOWNLOADS:
-        try:
-            url = next(sitemap_file_iterator)
-            if things_url is None:
-                # parse out the base things url from the first one we grab (they should all be the same)
-                things_url = pre_thing_host_url(url)
-                if things_url is not None:
-                    things_url += "/things"
+    while not constructed_all_futures_for_sitemap_file:
```

Comment:
The flow here has changed a bit -- previously we were reconstructing futures while servicing the event loop.  Now, we construct all the futures for a given sitemap file at one time.

There's a further optimization we could make where we could start fetching the next batch of things in the next sitemap file, but would require a bit more refactoring.